### DISCUSSION TITLE
Static website and dynamic routing. Is this the expected behaviour ?

### DISCUSSION THREAD

**Discussion by hlassiege** [nuxt/nuxt#16752] [Questions] [answered]
I have one question about static website and dynamic path
It works (surprisingly maybe) due to a side effect, but it works
I have a path like this one : /country/FR/hosts FR is dynamic.

When nuxt generate the website, I have
/country/index.html but of course, I don't have the other file because FR is dynamic

When it's deployed on cloudflare :
the browsers try to get /country/FR/hosts
cloudflare try to find the file, sends a 404 and redirect to the 404 page of Nuxt
the 404 is served, it is a nuxt application
Nuxt realize that he knows this URL and serve the url correctly
=> the page works, the user sees the final page. In the browser network tab, it is a 404

Not really a problem to be honest, I'm rather happy that it works
But I'd like to have your opinion on it. Is it what is expected ?

**Comment by TheAlexLichter** [comment, marked as answer]
Yes and no!

if you use dynamic SSR (i.e. via workers), this should **not happen** because the server is aware of routing and will resolve correctly.

If you have a static site (or SPA), there is (commonly) no way around it because the Web Server does not know that a route exists "inside Nuxt" and will serve the error document, which is the fallback file created by Nuxt 😋

**Reply by hlassiege** [reply]
thanks, it confirms our assumption. It works, but it's more a side effect ^^

**Reply by TheAlexLichter** [reply]
No no, it is intended to work like this - but the correct HTTP code can't be inferred sadly.
